icon_validator/rules/plugin_validators/version_bump_validator.py

```diff
@@ -0,0 +1,345 @@
+        # operates on dictionary of individual action/trigger/task
+        for input_key, input_value in local[SpecConstants.INPUT].items():
+            if input_value[SpecConstants.REQUIRED]:
+                if input_key not in remote[SpecConstants.INPUT] or \
+                  not remote[SpecConstants.INPUT][input_key][SpecConstants.REQUIRED]:
+                    raise ValidationException(f"Input has been added or changed to required in {input_key} without"
+                                              f" a major version increment."
+                                              f"{self.MAJOR_INSTRUCTIONS_STRING}")
+
+    def validate_no_output_no_longer_required(self, remote, local):
+        # verifies that outputs have not been changed from required to optional
+        if SpecConstants.OUTPUT in remote and SpecConstants.OUTPUT in local:
+            for output_key, output_vals in remote[SpecConstants.OUTPUT].items():
+                if SpecConstants.REQUIRED in output_vals and \
+                        SpecConstants.REQUIRED in local[SpecConstants.OUTPUT][output_key]:
+                    if output_vals[SpecConstants.REQUIRED]:
+                        # We know this output exists because this validator is called after verifying all outputs exist
+                        local_spec_req = local[SpecConstants.OUTPUT][output_key][SpecConstants.REQUIRED]
+                        if not local_spec_req:
+                            raise ValidationException(f"Output {output_key} has been changed to not required in "
+                                                      "without a major version increment."
+                                                      f"{self.MAJOR_INSTRUCTIONS_STRING}")
+
+    def validate_no_output_removed(self, remote, local):
+        for item in remote[SpecConstants.OUTPUT]:
+            if item not in local[SpecConstants.OUTPUT]:
+                raise ValidationException("Output has been removed from an action or trigger. This requires "
+                                          f"a major version increment.{self.MAJOR_INSTRUCTIONS_STRING}")
+
+    def validate_no_input_removed(self, remote, local):
+        for item in remote[SpecConstants.INPUT]:
+            if item not in local[SpecConstants.INPUT]:
+                raise ValidationException("Input has been removed from an action or trigger. This requires "
+                                          f"a major version increment.{self.MAJOR_INSTRUCTIONS_STRING}")
+
+    def validate_no_action_removed(self, remote, local):
+        for action in remote[SpecConstants.ACTIONS]:
+            if action not in local[SpecConstants.ACTIONS]:
+                raise ValidationException("Action has been removed from spec without a major version increment."
+                                          f"{self.MAJOR_INSTRUCTIONS_STRING}")
+
+    def validate_no_trigger_removed(self, remote, local):
+        for action in remote[SpecConstants.TRIGGERS]:
```

Comment:
This should be named `triggers` variable.

Reply:
done

Reply:
done